clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerMembershipManager.java

```diff
@@ -469,8 +498,14 @@ public int joinGroupEpoch() {
      */
     @Override
     public int leaveGroupEpoch() {
+        if (CloseOptions.GroupMembershipOperation.LEAVE_GROUP.equals(leaveGroupOperation)) {
+            return ConsumerGroupHeartbeatRequest.LEAVE_GROUP_MEMBER_EPOCH;
+        } else if (CloseOptions.GroupMembershipOperation.REMAIN_IN_GROUP.equals(leaveGroupOperation)) {
+            return ConsumerGroupHeartbeatRequest.LEAVE_GROUP_STATIC_MEMBER_EPOCH;
+        }
```

Comment:
why do we need to add this? IIUC the static member vs dynamic member leave group epoch should still only be based on whether the group instance id is set -- I assume if we do skip the leave group then we just wouldn't call this method to begin with (though I'm not super familiar with the "leave group epoch", just know it's part of the new rebalancing protocol)

Reply:
As https://github.com/apache/kafka/pull/17614#discussion_r1972859960 mentioned, consumerGroupHeartbeat always sends, so we need to select the "right" epoch to the server.